mono/mini/aot-runtime.c

```diff
@@ -5282,12 +5275,19 @@ mono_create_ftnptr_malloc (guint8 *code)
  * be called from mono_arch_init () during startup.
  */
 void
-mono_aot_register_jit_icall (const char *name, gpointer addr)
+mono_aot_register_jit_icall_info (MonoJitICallInfo *info, const char *name, gpointer addr)
 {
 	/* No need for locking */
-	if (!aot_jit_icall_hash)
-		aot_jit_icall_hash = g_hash_table_new (g_str_hash, g_str_equal);
-	g_hash_table_insert (aot_jit_icall_hash, (char*)name, addr);
+
```

Comment:
I suspect this entire function can be removed.

Reply:
Or change the switch to use the data. But I think remove the function and the initialization of the data. I need to check. It might also be possible to split the enum into a separate enum. This doesn’t seem horrible though.